internal/app/machined/pkg/runtime/v1alpha1/v1alpha1_sequencer_tasks.go

```diff
@@ -1998,3 +1998,37 @@ func pauseOnFailure(callback func(runtime.Sequence, interface{}) (runtime.TaskEx
 		}, name
 	}
 }
+
+func canLeaveEtcd() bool {
+	etcdID := (&services.Etcd{}).ID(nil)
+
+	services := system.Services(nil).List()
+
+	for _, service := range services {
+		if service.AsProto().Id != etcdID {
+			continue
+		}
+
+		//nolint:exhaustive
+		switch service.GetState() {
+		case events.StateRunning:
+			fallthrough
+		case events.StateStopping:
+			fallthrough
+		case events.StateFailed:
+			return true
+		}
+	}
+
+	return false
+}
+
+func canCordonNode() bool {
+	if _, err := os.Stat("/var/lib/kubelet/pki/kubelet-client-current.pem"); err != nil {
```

Comment:
cordoning might fail with other failures, e.g. if the API server is not responsive, so once again we should check more thoroughly ignoring the errors as needed

Reply:
I've made it ignore `Timeout` and `ServerTimeout` errors.